src/build/env.py

```diff
@@ -1,312 +1,313 @@
 
-    def __init__(self) -> None:
-        self._path: Optional[str] = None
 
-    def __enter__(self) -> IsolatedEnv:
-        """
-        Create an isolated build environment.
+def _create_isolated_env_venv(path: str) -> Tuple[str, str]:
+    """
+    :param path: venv path on disk
+    :returns: The Python executable and scripts folder
+    """
+    import venv
 
-        :return: The isolated build environment
-        """
-        self._path = tempfile.mkdtemp(prefix='build-env-')
-        try:
-            # use virtualenv when available (as it's faster than venv)
-            if _should_use_virtualenv():
-                self.log('Creating virtualenv isolated environment...')
-                executable, scripts_dir = _create_isolated_env_virtualenv(self._path)
-            else:
-                self.log('Creating venv isolated environment...')
-                executable, scripts_dir = _create_isolated_env_venv(self._path)
-            return _IsolatedEnvVenvPip(
-                path=self._path,
-                python_executable=executable,
-                scripts_dir=scripts_dir,
-                log=self.log,
-            )
-        except Exception:  # cleanup folder if creation fails
-            self.__exit__(*sys.exc_info())
-            raise
+    venv.EnvBuilder(symlinks=_fs_supports_symlinks(), with_pip=True).create(path)
+    return _get_isolated_env_executable_and_scripts_dir(path)
 
-    def __exit__(
-        self, exc_type: Optional[Type[BaseException]], exc_val: Optional[BaseException], exc_tb: Optional[TracebackType]
-    ) -> None:
-        """
-        Delete the created isolated build environment.
 
-        :param exc_type: The type of exception raised (if any)
-        :param exc_val: The value of exception raised (if any)
-        :param exc_tb: The traceback of exception raised (if any)
-        """
-        if self._path is not None and os.path.exists(self._path):  # in case the user already deleted skip remove
-            shutil.rmtree(self._path)
+def _create_isolated_env_virtualenv(path: str) -> Tuple[str, str]:
+    """
+    :param path: virtualenv path on disk
+    :returns: The Python executable and scripts folder
+    """
+    import virtualenv
 
-    @staticmethod
-    def log(message: str) -> None:
-        """
-        Prints message
+    cmd = [str(path), '--no-setuptools', '--no-wheel', '--activators', '']
+    result = virtualenv.cli_run(cmd, setup_logging=False)
+    executable = str(result.creator.exe)
+    script_dir = str(result.creator.script_dir)
+    return executable, script_dir
 
-        The default implementation uses the logging module but this function can be
-        overwritten by users to have a different implementation.
 
-        :param msg: Message to output
-        """
-        if sys.version_info >= (3, 8):
-            _logger.log(logging.INFO, message, stacklevel=2)
-        else:
-            _logger.log(logging.INFO, message)
+@_cache()
```

Comment:
This looks to me more cryptic than if we would just have put here the entire cache annotation.

Reply:
It's a backport of the `functools` function called `cache`, which works the same way.

Reply:
Ah, but then can we make it an actual backport, on 3.8/9 this exists let's use it from stdlib than our definition.

Reply:
Alright.

Reply:
Done.